.github/workflows/publish.yml

```diff
@@ -0,0 +1,81 @@
+name: Publish To Marketplace
+on: 
+  workflow_dispatch:
+    inputs:
+      releaseType:
+        description: 'Release type'
+        required: true
+        type: choice
+        default: 'pre-release'
+        options:
+          - pre-release
+          - release
+      updatedChangelog:
+        description: 'Reminder to Update CHANGELOG! (Not done by this workflow)'
+        type: boolean
+        required: true
+jobs:
+    release:
+      runs-on: windows-latest
+      env:
+        working-directory: ./extension/
+        RELEASETYPE: ${{ github.event.inputs.releaseType }}
+        VSCE_PAT: ${{ secrets.VSCE_PAT }}
+      steps:
+        - name: Changelog check
+          if: ${{ github.event.inputs.updatedChangelog == 'false' }}
+          shell: pwsh
+          run: |
+            Write-Error "Remember to update the changelog before creating a release!" -ErrorAction Stop
+        - uses: actions/checkout@v2
+        - uses: actions/setup-node@v2
+          with:
+            node-version: 16.x
+
+        - name: Git config
+          run: |
+            git config user.name 'github-actions[bot]'
+            git config user.email 'github-actions[bot]@users.noreply.github.com'
+            git config advice.addIgnoredFile false
+
+        - name: Install dependencies
+          run: |
+             npm ci
+             npm install -g vsce
+          working-directory: ${{env.working-directory}}
+
+        - name: "Package: ${{ env.RELEASETYPE }}"
```

Comment:
I think we've missed on thing...
This line in #338 :
`Run Pre-Release pipeline, but increase minor two steps, so a Pre-Release with a higher version than the release version always exists`

Check out `Release-ToMarketplace.ps1`, that is run by the manual release task.

Reply:
Solved this by adding a conditional step `Pre-release on release`.

Reply:
Result can be seen here https://github.com/theschitz/nab-al-tools/runs/5807636737?check_suite_focus=true

Reply:
I can't see the pre-release getting published, am I missing something?

Reply:
I assume that we need a conditional publishing step as well?

Reply:
You are correct, just my brain not working at a 100% :sweat_smile: